services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -240,8 +241,24 @@ default T mapToItem(Map<String, AttributeValue> attributeMap, boolean preserveEm
      * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map.
      *                    If set to false; null values in the Java object will be added as {@link AttributeValue} of
      *                    type 'nul' to the output map.
+     * @param requestConfiguration denotes the level of nesting, i.e. SHALLOW or NESTED, contained in the request params based
+     * which nested update query parameters would be dealt with
      * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
      */
+    Map<String, AttributeValue> itemToMap(T item, boolean ignoreNulls,
+                                          DynamoDBEnhancedRequestConfiguration requestConfiguration);
+
+    /**
+     * This is deprecated in favour of itemToMap(T item, boolean ignoreNulls, DynamoDBEnhancedRequestOverrideConfiguration
+     * ddbRequestOverrideConfiguration) This method takes a modelled object and converts it into a raw map of
+     * {@link AttributeValue} that the DynamoDb low-level SDK can work with.
+     *
+     * @param item        The modelled Java object to convert into a map of attributes.
+     * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map. If set to false;
+     *                    null values in the Java object will be added as {@link AttributeValue} of type 'nul' to the output map.
+     * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
+     */
+    @Deprecated
```

Comment:
We have discussed whether we should deprecate older APIs when there are new alternatives, since some applications break when deprecated attributes are detected. In this case I think it's warranted to deprecate, because itemToMap (older version) has a bug.

Reply:
Yep, I too went through the discussion points. I can probably call this out

Reply:
I'd like to understand this a bit more. Let's discuss this as a team in post standup